src/Core/Settings.h

```diff
@@ -532,6 +532,7 @@ static constexpr UInt64 operator""_GiB(unsigned long long value)
     M(UInt64, max_entries_for_hash_table_stats, 10'000, "How many entries hash table statistics collected during aggregation is allowed to have", 0) \
     M(UInt64, max_size_to_preallocate_for_aggregation, 10'000'000, "For how many elements it is allowed to preallocate space in all hash tables in total before aggregation", 0) \
     \
+    M(Bool, kafka_disable_num_consumers_limit, false, "Disable limit on kafka_num_consumers that depends on the number of available CPU cores", 0) \
```

Comment:
Does it worth to add an extra setting just for this? Maybe simply log a message with `error` or similar severity is enough?